spec/controllers/api/v1/tokens_controller_spec.rb

```diff
@@ -38,39 +38,52 @@
         end
       end
 
-      context 'given invalid user credentials' do
+      context 'given INCORRECT user credentials' do
         it 'returns bad_request status' do
           application = Fabricate(:application)
 
           params = {
             grant_type: 'password',
             email: 'dev@nimblehq.co',
-            password: '12345678'
+            password: 'incorrect'
           }.merge(oauth_application_params(application))
 
           post :create, params: params
 
           expect(response).to have_http_status(:bad_request)
         end
 
-        it 'returns an empty response' do
+        it 'returns an error message' do
           application = Fabricate(:application)
 
           params = {
             grant_type: 'password',
             email: 'dev@nimblehq.co',
-            password: '12345678'
+            password: 'incorrect'
           }.merge(oauth_application_params(application))
 
           post :create, params: params
 
-          response_body = JSON.parse(response.body)
+          expected_response = {
+            errors: [
+              {
+                code: 'invalid_email_or_password',
+                detail: 'Your email or password is incorrect. Please try again.',
+                source: {
+                  parameter: 'Doorkeeper::OAuth::Error'
```

Comment:
This reveals a bit too much of the internals of the application so it is not great in terms of security 🙈

Reply:
Yes, and I think the current `source` parameter is not correct, so I removed them in https://github.com/nimblehq/nimble-survey-web/pull/68/commits/aa70ecc36cabbeefdf714e391fdf90546010a622

Reply:
@olivierobert Removed them in https://github.com/nimblehq/nimble-survey-web/pull/68/commits/aa70ecc36cabbeefdf714e391fdf90546010a622